scrapy/pqueues.py

```diff
@@ -26,17 +26,8 @@ def _path_safe(text):
 
 
 class ScrapyPriorityQueue:
-    """A priority queue implemented using multiple internal queues (typically,
-    FIFO queues). It uses one internal queue for each priority value. The internal
-    queue must implement the following methods:
-
-        * push(obj)
-        * pop()
-        * close()
-        * __len__()
-
-    Optionally, the queue could provide a ``peek`` method, that should return the
-    next object to be returned by ``pop``, but without removing it from the queue.
+    """:ref:`Priority queue <priority-queues>` that sends requests based on the
+    :attr:`~scrapy.http.Request` priority value.
```

Comment:
Here we used to define (without much detail) a common interface for downstream queues (both memory downstream queues and disk downstream queues).

Now we link to the new topic, which describes the interface of disk downstream queues, but not of memory downstream queues. So in practice we are removing the documentation of the interface of memory downstream queues.

Maybe the new interface documentation was meant to cover both memory downstream queues and disk downstream queues? I see the interface changes seem to apply to both. If so, we need to rewrite the interface documentation a bit, and make it clear that disk-specific aspects (e.g. serialization and deserialization) only apply to disk queues.

Reply:
I removed mentions of disk from general context and add in context of serialization. I didn't touch state context, as memory queues with state have sense for me. E.g. you store requests in memory for performance, dump it to a disk at some moment and resume crawl from dumped state later.

Reply:
> I didn't touch state context, as memory queues with state have sense for me.

The implementation never passes `state` to memory queues, though. So it would be pointless to implement state handling in a custom downstream memory queue.